File: sdkwork-ai-sdk-go/api/knowledge_base.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type KnowledgeBaseApi struct {
    client *sdkhttp.Client
}

func NewKnowledgeBaseApi(client *sdkhttp.Client) *KnowledgeBaseApi {
    return &KnowledgeBaseApi{client: client}
}

// List documents
func (a *KnowledgeBaseApi) ListDocuments(knowledge_base_id string, query map[string]interface{}) (sdktypes.ListDocumentsGetResponse, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/knowledge-bases/%s/documents", knowledge_base_id)), query, nil)
    if err != nil {
        var zero sdktypes.ListDocumentsGetResponse
        return zero, err
    }
    return decodeResult[sdktypes.ListDocumentsGetResponse](raw)
}

// Add document
func (a *KnowledgeBaseApi) AddDocument(knowledge_base_id string, body sdktypes.DocumentAddRequest) (sdktypes.DocumentResponse, error) {
    raw, err := a.client.Post(AiApiPath(fmt.Sprintf("/knowledge-bases/%s/documents", knowledge_base_id)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.DocumentResponse
        return zero, err
    }
    return decodeResult[sdktypes.DocumentResponse](raw)
}

// Search knowledge
func (a *KnowledgeBaseApi) SearchKnowledge(body sdktypes.KnowledgeSearchRequest) (sdktypes.KnowledgeSearchResponse, error) {
    raw, err := a.client.Post(AiApiPath("/knowledge-bases/search"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.KnowledgeSearchResponse
        return zero, err
    }
    return decodeResult[sdktypes.KnowledgeSearchResponse](raw)
}

// List knowledge bases
func (a *KnowledgeBaseApi) ListKnowledgeBases(query map[string]interface{}) (sdktypes.KnowledgeBaseListResponse, error) {
    raw, err := a.client.Get(AiApiPath("/knowledge-bases"), query, nil)
    if err != nil {
        var zero sdktypes.KnowledgeBaseListResponse
        return zero, err
    }
    return decodeResult[sdktypes.KnowledgeBaseListResponse](raw)
}

// Create knowledge base
func (a *KnowledgeBaseApi) CreateKnowledgeBase(body sdktypes.KnowledgeBaseCreateRequest) (sdktypes.KnowledgeBaseResponse, error) {
    raw, err := a.client.Post(AiApiPath("/knowledge-bases"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.KnowledgeBaseResponse
        return zero, err
    }
    return decodeResult[sdktypes.KnowledgeBaseResponse](raw)
}

// Get document
func (a *KnowledgeBaseApi) GetDocument(knowledge_base_id string, document_id string) (sdktypes.DocumentResponse, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/knowledge-bases/%s/documents/%s", knowledge_base_id, document_id)), nil, nil)
    if err != nil {
        var zero sdktypes.DocumentResponse
        return zero, err
    }
    return decodeResult[sdktypes.DocumentResponse](raw)
}

// Delete document
func (a *KnowledgeBaseApi) DeleteDocument(knowledge_base_id string, document_id string) (struct{}, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/knowledge-bases/%s/documents/%s", knowledge_base_id, document_id)), nil, nil)
    if err != nil {
        var zero struct{}
        return zero, err
    }
    return decodeResult[struct{}](raw)
}

// Get knowledge base
func (a *KnowledgeBaseApi) GetKnowledgeBase(knowledge_base_id string) (sdktypes.KnowledgeBaseResponse, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/knowledge-bases/%s", knowledge_base_id)), nil, nil)
    if err != nil {
        var zero sdktypes.KnowledgeBaseResponse
        return zero, err
    }
    return decodeResult[sdktypes.KnowledgeBaseResponse](raw)
}

// Delete knowledge base
func (a *KnowledgeBaseApi) DeleteKnowledgeBase(knowledge_base_id string) (struct{}, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/knowledge-bases/%s", knowledge_base_id)), nil, nil)
    if err != nil {
        var zero struct{}
        return zero, err
    }
    return decodeResult[struct{}](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/knowledge-base.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { DocumentAddRequest, DocumentResponse, KnowledgeBaseCreateRequest, KnowledgeBaseListResponse, KnowledgeBaseResponse, KnowledgeSearchRequest, KnowledgeSearchResponse, ListDocumentsGetResponse } from '../types';


export class KnowledgeBaseApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** List documents */
  async listDocuments(knowledge_base_id: string | number, params?: QueryParams): Promise<ListDocumentsGetResponse> {
    return this.client.get<ListDocumentsGetResponse>(aiApiPath(`/knowledge-bases/${knowledge_base_id}/documents`), params);
  }

/** Add document */
  async addDocument(knowledge_base_id: string | number, body: DocumentAddRequest): Promise<DocumentResponse> {
    return this.client.post<DocumentResponse>(aiApiPath(`/knowledge-bases/${knowledge_base_id}/documents`), body);
  }

/** Search knowledge */
  async searchKnowledge(body: KnowledgeSearchRequest): Promise<KnowledgeSearchResponse> {
    return this.client.post<KnowledgeSearchResponse>(aiApiPath(`/knowledge-bases/search`), body);
  }

/** List knowledge bases */
  async listKnowledgeBases(params?: QueryParams): Promise<KnowledgeBaseListResponse> {
    return this.client.get<KnowledgeBaseListResponse>(aiApiPath(`/knowledge-bases`), params);
  }

/** Create knowledge base */
  async createKnowledgeBase(body: KnowledgeBaseCreateRequest): Promise<KnowledgeBaseResponse> {
    return this.client.post<KnowledgeBaseResponse>(aiApiPath(`/knowledge-bases`), body);
  }

/** Get document */
  async getDocument(knowledge_base_id: string | number, document_id: string | number): Promise<DocumentResponse> {
    return this.client.get<DocumentResponse>(aiApiPath(`/knowledge-bases/${knowledge_base_id}/documents/${document_id}`));
  }

/** Delete document */
  async deleteDocument(knowledge_base_id: string | number, document_id: string | number): Promise<void> {
    return this.client.delete<void>(aiApiPath(`/knowledge-bases/${knowledge_base_id}/documents/${document_id}`));
  }

/** Get knowledge base */
  async getKnowledgeBase(knowledge_base_id: string | number): Promise<KnowledgeBaseResponse> {
    return this.client.get<KnowledgeBaseResponse>(aiApiPath(`/knowledge-bases/${knowledge_base_id}`));
  }

/** Delete knowledge base */
  async deleteKnowledgeBase(knowledge_base_id: string | number): Promise<void> {
    return this.client.delete<void>(aiApiPath(`/knowledge-bases/${knowledge_base_id}`));
  }
}

export function createKnowledgeBaseApi(client: HttpClient): KnowledgeBaseApi {
  return new KnowledgeBaseApi(client);
}
